addon/-private/system/model/internal-model.js

```diff
@@ -583,10 +583,19 @@ export default class InternalModel {
     let changedKeys;
 
     if (this.hasRecord) {
-      changedKeys = this._changedKeys(data.attributes);
+      if (typeof this._record._changedKeys === 'function') {
```

Comment:
Discussed offline with @hjdivad, but I'd like to see these checks be done inside the `InternalModel.prototype._changedKeys` method itself. It would clean up some of this extra branching logic here and below, and also help prevent issues from future refactors that might add additional calls to `_changedKeys` but forget to do this `this._record._changedKeys` checking manually.